src/topology/separation.lean

```diff
@@ -150,6 +150,29 @@ instance subtype.t1_space {α : Type u} [topological_space α] [t1_space α] {p
 instance t1_space.t0_space [t1_space α] : t0_space α :=
 ⟨λ x y h, ⟨{z | z ≠ y}, is_open_ne, or.inl ⟨h, not_not_intro rfl⟩⟩⟩
 
+lemma t1_characterisation : t1_space α ↔
+  ∀ (x y), x ≠ y → (∃ (U : set α) (hU : is_open U), x ∈ U ∧ y ∉ U) :=
+begin
+  split;intro h,
+  { intros x y hxy,
+    exact ⟨{y}ᶜ,is_open_compl_iff.mpr (@t1_space.t1 α _inst_1 h y), mem_compl_singleton_iff.mpr hxy,
+      not_not.mpr rfl⟩},
+  { fconstructor,
```

Comment:
```suggestion
  split,
  { introI t1,
    intros x y hxy,
    exact ⟨{y}ᶜ, is_open_compl_iff.mpr (t1_space.t1 y),
            mem_compl_singleton_iff.mpr hxy,
            not_not.mpr rfl⟩},
  { intro h,
    fconstructor,
```

some reformatting; it feels kinda weird to `intro` together when it's different variables, IMO. Also, `introI` (and a bunch of [related tactics](https://leanprover-community.github.io/mathlib_docs/tactics.html#Instance%20cache%20tactics)) are helpful for type-class stuff, so you don't have to use `@t1_space.t1 ...`

Reply:
I've never heared about `introI`. Thanks, it will be very helpful in the future.